pylint/checkers/refactoring.py

```diff
@@ -261,13 +265,35 @@ def _check_redefined_argument_from_local(self, name_node):
                                  args=(name_node.name, ))
 
     @utils.check_messages('redefined-argument-from-local',
-                          'too-many-nested-blocks')
+                          'too-many-nested-blocks',
+                          'mutable-sequence-modified-in-loop')
     def visit_for(self, node):
         self._check_nested_blocks(node)
 
         for name in node.target.nodes_of_class(astroid.AssignName):
             self._check_redefined_argument_from_local(name)
 
+        try:
+            inferrend = node.iter.inferred().pop()
+            if (isinstance(node.iter, astroid.Name) and
+                    (isinstance(inferrend,
+                                type(astroid.util.Uninferable)) or
```

Comment:
You are referring to `type(astroid.util.Uninferable)` ?
This is required for sentences like `items = range(1, 6)`